src/code/z_parameter.c

```diff
@@ -16,124 +23,131 @@ typedef struct {
-    { SCENE_16GORON_HOUSE, 0x00, 0x00, 0x00 },
-    { SCENE_33ZORACITY, 0x00, 0x00, 0xC0 },
-    { SCENE_8ITEMSHOP, 0x00, 0x30, 0xC1 },
-    { SCENE_F01, 0x00, 0x00, 0x00 },
-    { SCENE_INISIE_BS, 0x00, 0x00, 0x00 },
-    { SCENE_30GYOSON, 0x00, 0x00, 0x00 },
-    { SCENE_31MISAKI, 0x00, 0x00, 0x00 },
-    { SCENE_TAKARAKUJI, 0x00, 0x30, 0xC1 },
-    { SCENE_UNSET_3A, 0x00, 0x00, 0x00 },
-    { SCENE_TORIDE, 0x00, 0x00, 0x00 },
-    { SCENE_FISHERMAN, 0x00, 0x30, 0xC1 },
-    { SCENE_GORONSHOP, 0x10, 0x30, 0xC1 },
-    { SCENE_DEKU_KING, 0x00, 0x30, 0xC0 },
-    { SCENE_LAST_GORON, 0x00, 0x3F, 0x00 },
-    { SCENE_24KEMONOMITI, 0x00, 0x00, 0x00 },
-    { SCENE_F01_B, 0x00, 0x30, 0x00 },
-    { SCENE_F01C, 0x00, 0x30, 0x00 },
-    { SCENE_BOTI, 0x00, 0x00, 0x00 },
-    { SCENE_HAKUGIN_BS, 0x00, 0x00, 0x00 },
-    { SCENE_20SICHITAI, 0x00, 0x00, 0x00 },
-    { SCENE_21MITURINMAE, 0x00, 0x00, 0x00 },
-    { SCENE_LAST_ZORA, 0x00, 0x3F, 0x00 },
-    { SCENE_11GORONNOSATO2, 0x00, 0x00, 0x00 },
-    { SCENE_SEA, 0x00, 0x00, 0x00 },
-    { SCENE_35TAKI, 0x00, 0x00, 0x00 },
-    { SCENE_REDEAD, 0x00, 0x00, 0x00 },
-    { SCENE_BANDROOM, 0x00, 0x30, 0xC0 },
-    { SCENE_11GORONNOSATO, 0x00, 0x00, 0x00 },
-    { SCENE_GORON_HAKA, 0x00, 0x00, 0x00 },
-    { SCENE_SECOM, 0x00, 0x3C, 0xC0 },
-    { SCENE_10YUKIYAMANOMURA, 0x00, 0x00, 0x00 },
-    { SCENE_TOUGITES, 0x00, 0x3F, 0xF0 },
-    { SCENE_DANPEI, 0x00, 0x30, 0x00 },
-    { SCENE_IKANAMAE, 0x00, 0x00, 0x00 },
-    { SCENE_DOUJOU, 0x00, 0x30, 0xC1 },
-    { SCENE_MUSICHOUSE, 0x00, 0x30, 0xC0 },
-    { SCENE_IKNINSIDE, 0x00, 0x3F, 0xC0 },
-    { SCENE_MAP_SHOP, 0x10, 0x30, 0xC1 },
-    { SCENE_F40, 0x00, 0x00, 0x00 },
-    { SCENE_F41, 0x00, 0x00, 0x00 },
-    { SCENE_10YUKIYAMANOMURA2, 0x00, 0x00, 0x00 },
-    { SCENE_14YUKIDAMANOMITI, 0x00, 0x00, 0x00 },
-    { SCENE_12HAKUGINMAE, 0x00, 0x00, 0x00 },
-    { SCENE_17SETUGEN, 0x00, 0x00, 0x00 },
-    { SCENE_17SETUGEN2, 0x00, 0x00, 0x00 },
-    { SCENE_SEA_BS, 0x00, 0x00, 0x00 },
-    { SCENE_RANDOM, 0x00, 0x00, 0x00 },
-    { SCENE_YADOYA, 0x10, 0x30, 0xC1 },
-    { SCENE_KONPEKI_ENT, 0x00, 0x00, 0x00 },
-    { SCENE_INSIDETOWER, 0x00, 0xFF, 0xC0 },
-    { SCENE_26SARUNOMORI, 0x00, 0x30, 0x00 },
-    { SCENE_LOST_WOODS, 0x00, 0x00, 0x00 },
-    { SCENE_LAST_LINK, 0x00, 0x3F, 0x00 },
-    { SCENE_SOUGEN, 0x00, 0x3F, 0x00 },
-    { SCENE_BOMYA, 0x10, 0x30, 0xC1 },
-    { SCENE_KYOJINNOMA, 0x00, 0x00, 0x00 },
-    { SCENE_KOEPONARACE, 0x00, 0x30, 0x00 },
-    { SCENE_GORONRACE, 0x00, 0x00, 0x00 },
-    { SCENE_TOWN, 0x00, 0x00, 0x00 },
-    { SCENE_ICHIBA, 0x00, 0x00, 0x00 },
-    { SCENE_BACKTOWN, 0x00, 0x00, 0x00 },
-    { SCENE_CLOCKTOWER, 0x00, 0x00, 0x00 },
-    { SCENE_ALLEY, 0x00, 0x00, 0x00 },
+    { SCENE_20SICHITAI2, SET_RESTRICTIONS(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0) },
+    { SCENE_UNSET_1, SET_RESTRICTIONS(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0) },
+    { SCENE_UNSET_2, SET_RESTRICTIONS(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0) },
```

Comment:
Another idea was to use `SET_RESTRICTIONS1` for flags1, `SET_RESTRICTIONS2` for flags2, and `SET_RESTRICTIONS3` for flags 3, and break them up into groups of 4 so it matches the struct better. But I started with only 1 `SET_RESTRICTIONS` with all arguments based on inspiration from the `GETITEM` table from z_player

Reply:
speaking of the player getitem table, what about
`{ SCENE_20SICHITAI2, SET_RESTRICTIONS(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0) },`
->
`RESTRICTIONS(SCENE_20SICHITAI2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),`
or `RESTRICTION_FLAGS` or whatever in line with `RestrictionFlags` :shrug:

Reply:
though tbh I think when I considered macroifying this in OoT I was more inclined to use flags to or `|` in into each field, it seemed more readable in my mind, but I never did get to it

Reply:
`{ SCENE_AYASHIISHOP, SET_RESTRICTIONS(0, 1, 0, 0, 0, 3, 0, 0, 3, 0, 0, 1) },` -> current
`RESTRICTIONS(SCENE_AYASHIISHOP, 0, 1, 0, 0, 0, 3, 0, 0, 3, 0, 0, 1)`, -> with scene in macro (agreed this is nice because you can also put the brackets into the macro, cleaning it up)
`RESTRICTIONS(SCENE_AYASHIISHOP, 0x10, 0x30, 0xC1)`, -> breaking it up into struct members
`RESTRICTIONS(SCENE_AYASHIISHOP, RESTRICTION_FLAGS1(0, 1, 0, 0), RESTRICTION_FLAGS2(0, 3, 0, 0), RESTRICTION_FLAGS3(3, 0, 0, 1))` -> makes it easier to parse and look up arguments
`RESTRICTIONS(SCENE_AYASHIISHOP, 1 << 4, 3 << 4, 3 << 8 | 1 << 0)`, -> using `|` in each field, with possible defines for 0/4/8/C)?

Is that last one what you were envisioning?

Reply:
In some way, it should pop out that you're setting each value to either 0, 1, 2, or 3, and somehow make it easy to map that to the particular struct member of restriction flags